graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/dynamic/QueryFetcher.java

```diff
@@ -43,18 +45,35 @@ public QueryFetcher(
   }
 
   @Override
-  protected Map<String, Object> get(
+  protected List<Map<String, Object>> get(
       DataFetchingEnvironment environment,
       DataStore dataStore,
       AuthenticationSubject authenticationSubject)
       throws UnauthorizedException {
     Keyspace keyspace = dataStore.schema().keyspace(model.getEntity().getKeyspaceName());
-    return querySingleEntity(
-        model.getEntity(),
-        environment.getArguments(),
-        Optional.of(model.getInputNames()),
-        dataStore,
-        keyspace,
-        authenticationSubject);
+
+    if (model.isReturnsList()) {
+      List<Map<String, Object>> listOfEntities =
```

Comment:
@olim7t I followed your suggestion from https://github.com/stargate/stargate/pull/709#discussion_r582431729 but there is one caveat here. Since in the case of a list is returned by the `queryListOfEntities`, we need to change this fetcher to return list (`List<Map<String, Object>>` cannot be the case to `Map<String, Object>`).
However, in that case, if the query returns one value, we need to wrap it in the `Collections.singletonList`.
I am not sure if this is a good solution for us. wdyt?

Reply:
The simplest solution is to have the fetcher return `Object`.

If that makes the code cleaner, we can indeed have two subclasses and have `QueryMappingModel.getDataFetcher` pick the right one depending on the value of `returnsList`.